src/utils/memoize.ts

```diff
@@ -0,0 +1,13 @@
+const _cache = new Map();
+export function memoize<T extends (...args: any[]) => any>(fn: T): T {
+  return function (...args: Parameters<T>): ReturnType<T> {
+    const key = JSON.stringify(args);
+    if (_cache.has(key)) {
+      return _cache.get(key);
+    }
+
+    const result = fn(...args);
+    _cache.set(key, result);
+    return result;
+  } as T;
+}
```

Comment:
We have a similar utility in `useMemoDeepEqual()`, but I just recently discovered it wasn't working quite like one would expect it to... I have a fix for that underway in `chore/instantation-to-hook` (which is growing quite large now).

Reply:
Ah yes, but this is hook depeendent, right? The `memoize` function is called outside a react-hook, that's why I made it :partying_face: